### PR TITLE
modified api and views of orders and checkout

### LINKED ISSUE #181: Design User Checkout Template

### LINKED ISSUE #191: Shop related
- [x] Cart
- [x] Search
- [x] Cashier and DEC
- [x] current shop in cart

### PR DESCRIPTION
**Pull request by HMMK99**

### CODE REVIEW

File: xshop/orders/api/views.py [hunk 1]
```diff
@@ -19,27 +17,7 @@ class OrderListCreateApi(APIView):
...
-            user.type[0]
-            in [
-                UserGroup.DATA_ENTRY_CLERK.title(),
-                UserGroup.CUSTOMER.title(),
-            ]
-        ):
-            orders = Order.objects.filter(user=user)
-        else:
-            return Response(status=status.HTTP_401_UNAUTHORIZED)
+        orders = Order.objects.all()
```
Comment (mmostafa74) [inline xshop/orders/api/views.py]: why do you removed all permissions?

File: xshop/orders/api/views.py [hunk 2]
```diff
@@ -62,27 +40,28 @@ class OrderDetailPatchApi(APIView):
...
-        order = get_object_or_404(Order, id=order_id)
-        if order.user != user or order.shop != user.shop or not user.is_superuser:
-            return Response(status=status.HTTP_401_UNAUTHORIZED)
-        serializer = self.serializer_class(order, data=request.data, partial=True)
-        serializer.is_valid(raise_exception=True)
-        updated_order = serializer.save()
-        return Response(self.serializer_class(updated_order).data)
+        try:
+            orders = Order.objects.get(id=order_id)
+            serializer = self.serializer_class(orders, data=request.data, partial=True)
```
Comment (mmostafa74) [inline xshop/orders/api/views.py]: same

File: xshop/orders/api/views.py [hunk 3]
```diff
@@ -62,27 +40,28 @@ class OrderDetailPatchApi(APIView):
         responses={200: OrderSerializer, 404: "Order not found"},
     )
     def get(self, request, order_id):
-        user = request.user
-        order = get_object_or_404(Order, id=order_id)
-        if not user.is_superuser and (order.user != user or order.shop != user.shop):
-            return Response(status=status.HTTP_401_UNAUTHORIZED)
-        serializer = self.serializer_class(order, many=False)
-        return Response(serializer.data)
+        try:
```
Comment (mmostafa74) [inline xshop/orders/api/views.py]: same

**Review by mmostafa74 — CHANGES_REQUESTED** [review]

File: xshop/orders/views.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            # serializer = self.serializer_class(data=request.data)
+            # serializer.is_valid(raise_exception=True)
+
+            # data = {"order_data": order.get_data}
+            # data["order_data"]["item_count"] = len(cart)
+
+            # data["order_data"]["full_price"] = str(full_price)
+            # data["address"] = serializer.validated_data.get("address")
+
+            return redirect("cart:cart_ops")
```
Comment (shahwan42) [inline xshop/orders/views.py:75]: Could you please redirect to a "Thank you" page that contains the order id  after payment?

**Review by shahwan42 — CHANGES_REQUESTED** [review]

**Comment by shahwan42** [comment]
Great job btw

**Pull request merged**